segmentation/utils/spatial_analysis.py

```diff
@@ -192,3 +197,62 @@ def calculate_cluster_spatial_enrichment(all_data, dist_mats, fovs=None,
         stats_xr = spatial_analysis_utils.calculate_enrichment_stats(close_num, close_num_rand)
         stats.loc[fovs[i], :, :] = stats_xr.values
     return values, stats
+
+
+def create_neighborhood_matrix(all_data, dist_matrices, fovs=None, distlim=50):
+    """Calculates the number of neighbor phenotypes for each cell
+
+        Args:
+            all_data: data for the all fov
+            dist_matrices: A dictionary that contains a cells x cells matrix with the euclidian
+                distance between centers of corresponding cells for every fov
+            fovs: patient labels to include in analysis. If argument is none, default is all labels used.
+            distlim: cell proximity threshold. Default is 50.
+        Returns:
+            cell_neighbor_counts: matrix with phenotype counts per cell
+            cell_neighbor_freqs: matrix with phenotype frequencies of
+                counts per phenotype/total phenotypes for each cell
+            cell_count: current cell in analysis"""
+
+    # Setup input and parameters
+    fov_col = "SampleID"
+    flowsom_col = "FlowSOM_ID"
+    cell_label_col = "cellLabelInImage"
+    cell_count = 0
+    if fovs is None:
+        fovs = sorted(list(set(all_data[fov_col])))
+
+    # Error Checking
+    if not np.isin(fovs, all_data[fov_col]).all():
+        raise ValueError("Points were not found in Expression Matrix")
+
+    # Subset just the sampleID, cellLabelInImage, and FlowSOMID
+    all_data = all_data[[fov_col, cell_label_col, flowsom_col]]
+    # Extract the columns with the cell phenotype codes
+    pheno_codes = all_data[flowsom_col].drop_duplicates()
+    # Get the total number of phenotypes
+    pheno_num = len(pheno_codes)
+
+    # initiate empty matrices for cell neighborhood data
+    cell_neighbor_counts = pd.DataFrame(np.zeros((all_data.shape[0], pheno_num + 2)))
+    cell_neighbor_freqs = pd.DataFrame(np.zeros((all_data.shape[0], pheno_num + 2)))
+
+    # Replace the first and second columns of cell_neighbor_counts with the fovs and cell labels respectively
+    cell_neighbor_counts[0] = all_data[fov_col]
+    cell_neighbor_counts[1] = all_data[cell_label_col]
+    cell_neighbor_freqs[0] = all_data[fov_col]
+    cell_neighbor_freqs[1] = all_data[cell_label_col]
```

Comment:
This is admittedly nitpicky to the extreme, but if you wanted, you could shorten this to something like:

cell_neighbor_counts[[0, 1]] = all_data[[fov_col, cell_label_col]]
cell_neighbor_freqs[[0, 1]] = all_data[[fov_col, cell_label_col]]

Reply:
also, you don't have to initiate `cell_neighbor_freqs` on line 238.  after you fill `cell_neighbor_counts` you can just copy it into a new `cell_neighbor_freqs`:
```
cell_neighbor_freqs = cell_neighbor_counts.copy()
```

sidenote, dataframe copies are deep by default, but if you want to be extra-extra sure, or make it so that people can see that in the code, you can call it with `.copy(deep=True)`